### ISSUE TITLE
why the ImList does not work form me?

### ISSUE THREAD

**Issue by Longhao307** [nus-cs2030/2122-s2#231]
Hello, I am trying the lab8. But i realise that the ImList doesnot work for me? is that the same for everyone? how could i resolve that?

**Comment by rexlionz** [comment]
hi longhao, what does not work mean? I would assume you meant that you tried using 
ImList = new ImList
the ImList code has changed so we are now unable to directly create an ImList using new if that is your issue.
Try to use ImList.of() instead! you can view the ImList file to see how to proceed. Hope this helps!

**Comment by wennwei** [comment]
hey! I ran into the same problem as well, the issue is that the new ImList provided requires the Pair class to compile and some of the methods in ImList such as ImList() are now private so you may have to use ImList.of() instead. Hope this helps!

**Comment by dinho2002** [comment]
Yeah, I had the same problem as well - Since new ImList<T>() doesn't work anymore because it is private in ImList.java you can use ImList.of() for creating an empty ImList.

**Comment by jlee1007** [comment]
for compilation u need to compile Pair.java than ImList.java in this order
for instantiating because ImList is now private, you must use ImList<T>.of() instead of new ImList

**Comment by yizziepeasy** [comment]
Hello, you can try this order `jshell Pair.java ImList.java ..... (other .java files)`. If this doesn't resolve your issue, do let me know!

**Comment by alwinngjw** [comment]
Hi, using ImList.of() should solve your problem. 
We are unable to use "new ImList()" anymore as the constructor is now private.

Also when testing your code in jshell, try to /open Pair.java first before ImList.java as ImList.java has a dependency on Pair.java

Hope this helps

**Comment by Darien2307** [comment]
> I think another potential reason is that the provided ImList.java for this Lab 8 has an incorrect indentation which failed the check style and leads to failing codecrunch test cases. I encountered this and I had to just indent the part that says the indentation has an error. Hope this helps!

Yup, I also had this issue but its okay not to submit your Pair and ImList files onto Codecrunch. It will still grade your other files normally.

**Comment by StanleyMak** [comment]
Yes, do check the new ImList or Pair classes that we are given as the old ImList class we have been using has public contructors. now they are set to private so you have to use ImList.of()

**Comment by lindyapeintern** [comment]
yeap just do jshell on pair and imlist.java and it should work!

**Comment by Hamayah** [comment]
Yup it's super important to look at the new methods in ImList!
